service/service_test.go

```diff
@@ -335,6 +335,9 @@ func TestClose(t *testing.T) {
 				time.Sleep(100 * time.Millisecond)
 				return nil
 			},
+			CloseFunc: func(ctx context.Context, optFuncs ...kafka.OptFunc) error {
```

Comment:
Lack of this was causing a unit test failure after upgrading to go1.21 (presumably timing related due to concurrent tests)